src/rpc/client.rs

```diff
@@ -373,14 +374,14 @@ impl RpcClient {
                     .map_ok(|(region, client)| {
                         (scan, region.range(), RawContext::new(region, client, cf))
                     })
-                    .and_then(|(mut scan, region_range, context)| {
-                        let (start_key, end_key) = scan.range();
+                    .and_then(|(scan, region_range, context)| {
+                        let (ref start_key, ref end_key) = scan.range;
```

Comment:
Any way we can take here to avoid the clone?

Reply:
I think we can, but it requires some significant refactoring of `ScanRegionsContext`. That will clash quite a lot with https://github.com/tikv/client-rust/pull/70, so I'd prefer to leave it until after we land both these PRs. I would like to do that refactoring though.